api/jobs.go

```diff
@@ -36,11 +36,33 @@ type Jobs struct {
 	client *Client
 }
 
+// JobsParseRequest is used for arguments of the /vi/jobs/parse endpoint
+type JobsParseRequest struct {
+	//JobHCL is an hcl jobspec
+	JobHCL string
+
+	//Canonicalize is a flag as to if the server should return default values
+	//for unset fields
+	Canonicalize bool
+}
+
 // Jobs returns a handle on the jobs endpoints.
 func (c *Client) Jobs() *Jobs {
 	return &Jobs{client: c}
 }
 
+// Parse is used to convert the HCL repesentation of a Job to JSON server side.
+// To parse the HCL client side see package github.com/hashicorp/nomad/jobspec
+func (j *Jobs) Parse(jobHCL string, canonicalize bool) (*Job, error) {
```

Comment:
Name it `ParseHCL` so we are futureproofed if we want to add a `ParseJSON` with canonicalize for people who want to just have defaults set.